Additions in Java classes architecture

Alvahaze opened the pull request:
- Additions in Java classes architecture  aa62507
- Additions in UML diagram  f01ef67
- Small fix in UML  659bedf
- Small fix in Java class  783d6ce
- Fix in classes  b770fbf
- Getters fix  25810e4
- Storage fix  3223aa2
- Trader fix  f204dbd

pisarevski committed:
Additions in Java classes architecture

- Add interface IValuable
- Add interface IMagnetable
- Add class Sheep
- Add class Shrapnel
- ShipPart class now implements IValuable interface

PunyDevil committed:
Additions in UML diagram

- Add package stuff
- Add class Sheep
- Add class Shrapnel
- Add interface IValuable
- Add interface IMagnetable
- ShipPart class now implements IValuable interface

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -172,6 +171,24 @@ package rape.brutal.gitstarship {
 
+    IValuable <|.. ShipPart
+
+    interface IValuable {
```
Why Gun isn't `IValuable`? We can't buy a gun?

ViRGiL175 reviewed (changes requested):
UML is wrong.

ViRGiL175 commented:
Reviewed.

pisarevski committed:
Small fix in UML

- Class Gun now implements IValuable interface

pisarevski committed:
Small fix in Java class

- Class Gun now implements IValuable interface

pisarevski committed:
Fix in classes

- Railgun, Turret, Engine, Hull now implements method getCost

ViRGiL175 reviewed (changes requested):
Not compilable.

ViRGiL175 commented:
Reviewed.

ViRGiL175 committed:
Storage fix

- Full working Java method for Storage class
- UML method addition

ViRGiL175 committed:
Trader fix

- Protected fields for Trader base class
- Corresponding UML changes

pisarevski committed:
Getters fix

- Getters now returns field value instead of zero or null
- Storage now is protected

ViRGiL175 reviewed (approved):
Good work.

ViRGiL175 commented:
Approved.
